### PR TITLE
`sonobuoy generate` first pass

### PR DESCRIPTION
**Pull request by liztio**
I don't know how to test this because seemingly nothing can handle multiple YAML documents?

### CODE REVIEW

File: cmd/sonobuoy/app/utils/mode/mode.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
+package mode
```
Comment (chuckha) [inline cmd/sonobuoy/app/utils/mode/mode.go]: duplicate package 📦 📦

Comment (chuckha) [reply cmd/sonobuoy/app/utils/mode/mode.go]: I'd also be ok dropping utils and adding mode/ as a directory under app/, but up to you

File: cmd/sonobuoy/app/operations/gen.go [hunk 2]
```diff
@@ -17,20 +17,52 @@ limitations under the License.
...
 type GenConfig struct {
-	Path string
-	opts RunConfig
+	Path     string
+	ModeName mode.Name
+	Image    string
+}
+
+type templateValues struct {
+	E2EFocus       string
```
Comment (timothysc) [inline cmd/sonobuoy/app/operations/gen.go]: we'll need skip and provider too.  

Ideally, I was hoping gen/run could use @chuckha's tooling to query the cluster and get the provider settings and fill it in properly for us.

Comment (liztio) [reply cmd/sonobuoy/app/operations/gen.go]: Neither of those are used by any of the examples right now. What's skip and provider for?

Comment (timothysc) [reply cmd/sonobuoy/app/operations/gen.go]: skip is the regex list passed to the image, provider right now is plumbing but will be leveraged a lot more in the not-to-distant future.  

We get asked all the time by folks customizing the focus, and skip vars... 
https://github.com/heptio/kube-conformance/blob/master/Dockerfile#L32

**Pull request merged**